src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerWorkspace.cs

```diff
@@ -67,7 +67,14 @@ ValueTask ILspWorkspace.UpdateTextIfPresentAsync(DocumentId documentId, SourceTe
         return this.ProjectSystemProjectFactory.ApplyChangeToWorkspaceAsync(
             _ =>
             {
-                this.OnDocumentTextChanged(documentId, sourceText, PreservationMode.PreserveIdentity, requireDocumentPresent: false);
+                if (CurrentSolution.ContainsDocument(documentId))
+                {
+                    this.OnDocumentTextChanged(documentId, sourceText, PreservationMode.PreserveIdentity, requireDocumentPresent: false);
+                }
+                else if (CurrentSolution.ContainsAdditionalDocument(documentId))
+                {
+                    this.OnAdditionalDocumentTextChanged(documentId, sourceText, PreservationMode.PreserveIdentity);
```

Comment:
if we were to do this, i would expect teh requireDocumentPresent: false flag as well.

Reply:
So, are we ok with changing the OnAdditionalDocument* APIs?

Reply:
Done